subprojects/publish/src/main/java/org/gradle/api/publish/PublishRetrier.java

```diff
@@ -14,10 +14,22 @@
  * limitations under the License.
  */
 
-package org.gradle.api.publication.maven.internal.action;
+package org.gradle.api.publish;
 
-import org.sonatype.aether.deployment.DeploymentException;
+import org.gradle.api.Incubating;
 
-interface MavenDeployRetrier {
-    void deployWithRetry() throws DeploymentException;
+/**
+ * Publisher that is able to retry in case of intermittent network issues.
+ *
+ * @since 6.3
+ */
+@Incubating
+public interface PublishRetrier {
```

Comment:
:x: This shouldn't be public API IMHO. Any reason why you put this on the public API?

Reply:
Addressed in https://github.com/gradle/gradle/pull/12024/commits/700fac49f84913d6f73d6b87226c6645da89c029